mantidproject/mantid
# Add mantidworkbench as an entry point for the mantidworkbench Conda package

### Description of work
- Add `mantidworkbench` as an entry point to the application. Users can now launch with either `workbench` or `mantidworkbench` from within a Conda environment.
- Configure the Conda package entry points to use `jemalloc` on Linux. Previously, the `workbench` entry point did not include the same `jemalloc` setup as the `mantidworkbench` script that was available on Linux installs. Now on Linux both entry points will call the same launch script that will configure `jemalloc` to be used.
Here is a diagram to help illustrate the changes:
![image](https://github.com/user-attachments/assets/b4a3df23-a557-4194-a9df-70622ace6668)


#### Summary of work
Added a new wrapper script called `workbench_launch_wrapper` that is now the entry point defined in the conda-build recipe. The new script will detect whether you are using Linux. If not, workbench is launched as before. If it is Linux, it will call the script that used to be named `mantidworkbench` that will configure `jemalloc` before launching the application.

#### Purpose of work
- There have been cases where users have asked for a `mantidworkbench` entry point to avoid confusion.
- It is already an entry point on Linux, but will perform additional steps before launching the application (e.g. setting up `jemalloc`)
- Currently if you launch on Linux with the Conda entry point (`workbench`), it bypasses the `jemalloc` setup, and the application does not perform well.

Fixes #37538 

### To test:
On **all OS**:
Install the test package:
`conda install jclarkestfc/label/entryPointTest::mantidworkbench`
Test the following things:
1. Verify that mantid workbench launches with both `mantidworkbench` and `workbench`
2. Test that the error report functions with both entry points (e.g. use the SegFault algorithm)
3. Test that the error reporter doesn't pop up when passing the `--single-process` argument to either entry point
4. (Linux only) verify that both launch entry points configure jemalloc correctly by running `os.environ['LD_PRELOAD']` in workbench's python console. It should output the location of `libjemalloc.so.2` from inside your Conda environment.
5. (Linux only) Check that the --debug argument works when launching with either entry point.
6. On each OS, install the standalone packages and make sure they still work as expected:
https://builds.mantidproject.org/job/build_packages_from_branch/957/





---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
